Sources/Swift/Tools/SentryLog.swift

```diff
@@ -30,7 +35,13 @@ class SentryLog: NSObject {
      */
     @objc
     static func willLog(atLevel level: SentryLevel) -> Bool {
-        return isDebug && level != .none && level.rawValue >= diagnosticLevel.rawValue
+        if level == .none {
+            return false
+        }
+        if level.rawValue >= alwaysLevel.rawValue {
+            return true
+        }
```

Comment:
l: Can we change this to `guard` to ensure early return in compile time?

Reply:
As much as I like the `guard` statement, I am not sure if the compromise in readability is worth it:

```suggestion
        guard level != .none else {
            return false
        }
        guard level.rawValue < alwaysLevel.rawValue else {
            return true
        }
```